File: 0049-group-anagrams/0049-group-anagrams.ts
function groupAnagrams(strs: string[]): string[][] {
    const groups = new Map();
    for (const string of strs){
        const key = string.split('').sort().join('');
        if (!groups.has(key)) groups.set(key, [])
        groups.get(key).push(string);
    }
    return Array.from(groups.values());
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        groups = {}
        for string in strs:
            key = "".join(sorted(string))
            if key not in groups:
                groups[key] = []
            sorted_word = "".join(sorted(string))
            groups[key].append(string)
        return list(groups.values())
        